packages-rs/drainpipe/src/main.rs

```diff
@@ -10,16 +10,28 @@ mod firehose;
 mod schema;
 
 #[derive(Debug)]
-enum ProcessError {
-    DecodeError { _e: firehose::Error },
-    ProcessError { seq: i64, error: anyhow::Error },
+enum ProcessErrorKind {
+    DecodeError,
+    ProcessError,
+}
+
+#[derive(Debug)]
+struct ProcessError {
+    seq: i64,
+    inner: anyhow::Error,
+    msg: Vec<u8>,
+    kind: ProcessErrorKind,
 }
```

Comment:
This is the refactor mentioned in the description.  It lets us remove the `if` statement that was on [line 146](https://github.com/likeandscribe/unravel/pull/74/files#diff-6494e843f601dfc44174e4353fd7b836f5f2fb15a51868fae03256705269c5f6L146).